apps/mobile-wallet/src/features/modals/BottomModal2.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  BottomSheetBackdropProps,
+  BottomSheetFlashList,
+  BottomSheetModal,
+  BottomSheetModalProps,
+  BottomSheetScrollView,
+  BottomSheetView
+} from '@gorhom/bottom-sheet'
+import { BottomSheetFlashListProps } from '@gorhom/bottom-sheet/lib/typescript/components/bottomSheetScrollable/BottomSheetFlashList'
+import { useCallback, useEffect, useRef } from 'react'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { useTheme } from 'styled-components/native'
+
+import BottomModalBackdrop from '~/features/modals/BottomModalBackdrop'
+import { BottomModalBaseProps } from '~/features/modals/BottomModalBase'
+import BottomModalHandle from '~/features/modals/BottomModalHandle'
+import BottomModalHeader from '~/features/modals/BottomModalHeader'
+import { closeModal, removeModal } from '~/features/modals/modalActions'
+import { useAppDispatch } from '~/hooks/redux'
+import { DEFAULT_MARGIN, VERTICAL_GAP } from '~/style/globalStyle'
+
+export type BottomModal2Props<T> = BottomModalWithChildrenProps | BottomModalFlashListProps<T>
+
+interface BottomModalWithChildrenProps extends BottomModalBaseProps {
+  notScrollable?: boolean
+  bottomSheetModalProps?: Omit<BottomSheetModalProps, 'children'>
+}
+
+interface BottomModalFlashListProps<T> extends Omit<BottomModalBaseProps, 'children'> {
+  flashListProps?: BottomSheetFlashListProps<T>
+  bottomSheetModalProps?: Omit<BottomSheetModalProps, 'children'>
+}
+
+const BottomModal2 = <T,>(props: BottomModal2Props<T>) => {
+  const bottomSheetModalRef = useRef<BottomSheetModal>(null)
+  const dispatch = useAppDispatch()
+  const theme = useTheme()
+  const safeAreaInsets = useSafeAreaInsets()
+
+  useEffect(() => {
+    bottomSheetModalRef.current?.present()
+  }, [])
+
+  const handleClose = useCallback(() => {
+    bottomSheetModalRef.current?.dismiss()
+  }, [])
+
+  const handleDismiss = useCallback(() => {
+    dispatch(closeModal({ id: props.modalId }))
+    dispatch(removeModal({ id: props.modalId }))
+  }, [dispatch, props.modalId])
```

Comment:
When the sheet gets dismissed (either by clicking the backdrop, or pulling down, or programmatically) we can completely remove the modal from our Redux state directly since we do not handle the animations ourselves anymore.

Reply:
That's so good! But it means we could probably simplify our Redux actions by only keeping `closeModal`, no?